# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: /home/hal/NetBeansProjects/smartmontools-code/smartmontools
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: ataprint.cpp
--- ataprint.cpp Base (BASE)
+++ ataprint.cpp Locally Modified (Based On LOCAL)
@@ -802,8 +802,6 @@
   time_t now = time(0);
   char timedatetz[DATEANDEPOCHLEN]; dateandtimezoneepoch(timedatetz, now);
   jout("Local Time is:    %s\n", timedatetz);
-  jglb["local_time"]["time_t"] = now;
-  jglb["local_time"]["asctime"] = timedatetz;

   // Print warning message, if there is one
   if (dbentry && *dbentry->warningmsg)
Index: nvmeprint.cpp
--- nvmeprint.cpp Base (BASE)
+++ nvmeprint.cpp Locally Modified (Based On LOCAL)
@@ -202,10 +202,8 @@
   }

   time_t now = time(0);
-  char td[DATEANDEPOCHLEN]; dateandtimezoneepoch(td, now);
-  jout("Local Time is:                      %s\n", td);
-  jglb["local_time"]["time_t"] = now;
-  jglb["local_time"]["asctime"] = td;
+  char timedatetz[DATEANDEPOCHLEN]; dateandtimezoneepoch(timedatetz, now);
+  jout("Local Time is:                      %s\n", timedatetz);
 }

 // Format scaled power value.
Index: scsiprint.cpp
--- scsiprint.cpp Base (BASE)
+++ scsiprint.cpp Locally Modified (Based On LOCAL)
@@ -2135,8 +2135,6 @@
     time_t now = time(0);
     char timedatetz[DATEANDEPOCHLEN]; dateandtimezoneepoch(timedatetz, now);
     jout("Local Time is:        %s\n", timedatetz);
-    jglb["local_time"]["time_t"] = now;
-    jglb["local_time"]["asctime"] = timedatetz;

     // See if unit accepts SCSI commands from us
     if ((err = scsiTestUnitReady(device))) {
Index: smartctl.cpp
--- smartctl.cpp Base (BASE)
+++ smartctl.cpp Locally Modified (Based On LOCAL)
@@ -1653,6 +1653,11 @@
     if (jglb.has_uint128_output())
       jglb["smartctl"]["uint128_precision_bits"] = uint128_to_str_precision_bits();
     jglb["smartctl"]["exit_status"] = status;
+    // print current time and date and timezone
+    time_t now = time(0);
+    char timedatetz[DATEANDEPOCHLEN]; dateandtimezoneepoch(timedatetz, now);
+    jglb["local_time"]["time_t"] = now;
+    jglb["local_time"]["asctime"] = timedatetz;
     jglb.print(stdout, print_as_json_options);
   }
   catch (const std::bad_alloc & /*ex*/) {
